lib/client-assets.php

```diff
@@ -1206,33 +1206,14 @@ function gutenberg_editor_scripts_and_styles( $hook ) {
 	// Initialize the editor.
 	$gutenberg_theme_support = get_theme_support( 'gutenberg' );
 	$align_wide              = get_theme_support( 'align-wide' );
-	$color_palette           = (array) get_theme_support( 'editor-color-palette' );
+	$color_palette           = current( (array) get_theme_support( 'editor-color-palette' ) );
 	$font_sizes              = current( (array) get_theme_support( 'editor-font-sizes' ) );
 
-	// Backcompat for Color Palette set as multiple parameters.
-	if ( isset( $color_palette[0] ) && ( is_string( $color_palette[0] ) || isset( $color_palette[0]['color'] ) ) ) {
-		_doing_it_wrong(
-			'add_theme_support()',
-			__( 'Setting colors using multiple parameters is deprecated. Please pass a single parameter with an array of colors. See https://wordpress.org/gutenberg/handbook/extensibility/theme-support/ for details.', 'gutenberg' ),
-			'3.4.0'
-		);
-	} else {
-		$color_palette = current( $color_palette );
-	}
-
 	// Backcompat for Color Palette set through `gutenberg` array.
 	if ( empty( $color_palette ) && ! empty( $gutenberg_theme_support[0]['colors'] ) ) {
 		$color_palette = $gutenberg_theme_support[0]['colors'];
 	}
 
-	if ( ! empty( $gutenberg_theme_support ) ) {
```

Comment:
@jorgefilipecosta, I think I missed something here, can you help?

Reply:
Hi @gziolo, this code is to keep compatibility with the old Gutenberg theme support. The code block `// Backcompat for Color Palette set through `gutenberg` array.` should also be removed.
I also did a checkup on the logic and we had missed a removal in a previous version so I removed that code. I added my changes in a new commit https://github.com/WordPress/gutenberg/pull/8276/commits/417203c04d884d90a875172d1cc1e1ed7171f4eb.